src/relations/postgresql_provider.py

```diff
@@ -115,11 +112,8 @@ def _on_database_requested(self, event: DatabaseRequestedEvent) -> None:
             # Set the database name
             self.database_provides.set_database(event.relation.id, database)
 
-            # Set connection string URI.
-            self.database_provides.set_uris(
-                event.relation.id,
-                f"postgresql://{user}:{password}@{self.charm.primary_endpoint}:{DATABASE_PORT}/{database}",
-            )
```

Comment:
`_on_database_requested` is ran only when the db is requested, so the uri won't be updated if there's a switchover.